internal/cloud/realm/auth.go

```diff
@@ -76,18 +76,18 @@ func (c *client) AuthProfile() (AuthProfile, error) {
 func (c *client) getAuthToken(options api.RequestOptions) (string, error) {
 	session := c.authService.Session()
 
-	if !options.NoAuth {
-		if session.AccessToken == "" {
+	if options.RefreshAuth {
+		if session.RefreshToken == "" {
 			return "", ErrInvalidSession{}
 		}
-		return session.AccessToken, nil
+		return session.RefreshToken, nil
 	}
 
-	if options.RefreshAuth {
-		if session.RefreshToken == "" {
+	if !options.NoAuth {
+		if session.AccessToken == "" {
 			return "", ErrInvalidSession{}
 		}
-		return session.RefreshToken, nil
+		return session.AccessToken, nil
 	}
```

Comment:
This diff is a bit noisy, but I actually just moved one small block in front of another